src/command_modules/azure-cli-sql/tests/test_sql_commands.py

```diff
@@ -58,8 +60,12 @@ def _get_resource_group(self, **kwargs):
 class SqlServerMgmtScenarioTest(ScenarioTest):
 
     @ResourceGroupPreparer()
-    def test_sql_server_mgmt(self, resource_group, resource_group_location):
-        servers = ['cliautomation53', 'cliautomation54']  # TODO: Server names should be randomized
+    # Skip create in SqlServerPreparer because we want to create in the test. So why even use
+    # SqlServerPreparer? We still want to generate random server name and replace it in the test
+    # recording, which SqlServerPreparer does very nicely.
```

Comment:
For this cause, you don't need a preparer. The `create_random_name` method if `ScenarioTest` will register the random name with moniker in a name replacer. The replacer to alter the recording file.

Reference: https://github.com/Azure/azure-cli/blob/master/src/azure-cli-testsdk/azure/cli/testsdk/base.py#L173

Give it a try.